common/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -413,6 +414,28 @@ public static void register(int id, Writeable.Reader<DataType<?>> dataType) {
         return streamer;
     }
 
+    public static boolean compareTypesById(DataType<?> left, DataType<?> right) {
+        if (left.id() != right.id()) {
+            return false;
+        } else if (isArray(left)) {
+            return compareTypesById(unnest(left), unnest(right));
```

Comment:
Is it okay to use `unnest` like this here? What if one array has more dimensions than the other?